Fix following issues:

LinHungWu opened the pull request:
Three JVMs and 1 webserver, that are all part of 2 groups. Click Start All Groups. It issues 2 starts for each JVM.
Three JVMs and 1 webserver, that are all part of 2 groups. Click Stop All Groups. It issues 2 stops for each JVM, giving nasty error messages in the log.
Three JVMs and 1 webserver, that are all part of 2 groups. The JVMs are up. Click Start All Groups. It issues starts for the running JVMs.

Before you contribute, please review these guidelines to help ensure a smooth process for everyone.

Thanks.

* Read [how to properly contribute to open source projects on Github][1].
* Fork the project.
* Use a feature branch.
* Write [good commit messages][2].
* Use the same coding conventions as the rest of the project.
* Commit locally and push to your fork until you are happy with your contribution.
* Make sure to add tests and verify all the tests are passing when merging upstream.
* Add an entry to the [Changelog][3] accordingly.
* Please add your name to the [CONTRIBUTORS.md][7] file. Adding your name to the [CONTRIBUTORS.md][7] file signifies agreement to all rights and reservations provided by the [License][4].
* [Squash related commits together][5].
* Open a [pull request][6].
* The pull request will be reviewed by the community and merged by the project committers.

[1]: http://gun.io/blog/how-to-github-fork-branch-and-pull-request
[2]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[3]: ./CHANGELOG.md
[4]: ./LICENSE
[5]: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
[6]: https://help.github.com/articles/using-pull-requests
[7]: ./CONTRIBUTORS.md

LinHungWu committed:
Fix following issues:
Three JVMs and 1 webserver, that are all part of 2 groups. Click Start All Groups. It issues 2 starts for each JVM.
Three JVMs and 1 webserver, that are all part of 2 groups. Click Stop All Groups. It issues 2 stops for each JVM, giving nasty error messages in the log.
Three JVMs and 1 webserver, that are all part of 2 groups. The JVMs are up. Click Start All Groups. It issues starts for the running JVMs.

LinHungWu committed:
Fix following issues:
Three JVMs and 1 webserver, that are all part of 2 groups. Click Start All Groups. It issues 2 starts for each JVM.
Three JVMs and 1 webserver, that are all part of 2 groups. Click Stop All Groups. It issues 2 stops for each JVM, giving nasty error messages in the log.
Three JVMs and 1 webserver, that are all part of 2 groups. The JVMs are up. Click Start All Groups. It issues starts for the running JVMs.

jmahmoodcc committed:
add transactional annotation so group is returned with JVMs

jeddcuison committed:
Make delete web server in configurations tab delete only new web servers

jeddcuison committed:
Make delete JVM in configurations tab delete only new JVMs

jeddcuison committed:
Return status 204 since we're not returning any content after web server deletion

jmahmoodcc committed:
make application deploy conf transaction readonly=true

akinny commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImpl.java:
```diff
@@ -23,6 +23,7 @@
     private final GroupService groupService;
     private final JvmControlService jvmControlService;
     private final ExecutorService executorService;
+    private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger(GroupJvmControlServiceImpl.class);
```
Import class

PeterTeunissen replied:
+1

LinHungWu replied:
will fix

jeddcuison commented on jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupWebServerControlServiceImplTest.java:
```diff
@@ -5,6 +5,7 @@
 import com.cerner.jwala.common.domain.model.user.User;
```
Can we write a test case for this fix ?

LinHungWu replied:
yes, will do.

PeterTeunissen commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImpl.java:
```diff
@@ -53,19 +54,25 @@ public void controlAllJvms(final ControlGroupJvmRequest controlGroupJvmRequest,
             }
         }
-
+        LOGGER.info("jvm size: " + jvms.size());
```
Need more detail what you are logging here (or remove...). If I see this in a log: what does it mean?

LinHungWu replied:
I will put more meaningful log.

PeterTeunissen commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupWebServerControlServiceImpl.java:
```diff
@@ -24,6 +24,7 @@
     private final GroupService groupService;
     private final WebServerControlService webServerControlService;
     private final ExecutorService executorService;
+    private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger(GroupWebServerControlServiceImpl.class);
```
Import the package/class.

LinHungWu replied:
will fix.

jeddcuison commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupWebServerControlServiceImpl.java:
```diff
@@ -59,13 +60,19 @@ public void controlAllWebSevers(final ControlGroupWebServerRequest controlGroupW
-                }
-            });
+            if ("START".equalsIgnoreCase(controlGroupWebServerRequest.getControlOperation().name()) && "STARTED".equalsIgnoreCase(webServer.getState().toStateLabel())) {
+                LOGGER.info("webServer {} already in state {}.", webServer.getName(), webServer.getState().toStateLabel());
```
Do we really need this ? I mean applying conditions when logging ? I guess we used this for trouble shooting and you wanted to limit the logs to specific criteria but it's not a good practice in my opinion because you will miss out on some of the log data e.g. maybe someday you would need logging on the other web server name and state combinations but since you have a condition here, then you would have to change the code to see those. In addition, we can use tools to narrow down the logs that you are analyzing instead of having our application do that.

LinHungWu replied:
I create a validate method with unit test now.

jeddcuison reviewed (changes requested):
- I'm not too keen on writing code just to control logging. If your purpose is to make it easier to analyze, there are viewers for that
- Can we write a test to verify this fix

Sharvaribarve committed:
Overrid functions in certain cases for specific rules and wrote unit tests

Sharvaribarve committed:
Merge branch 'APISpecialCharacters' into RESTSpecialCharacters

jeddcuison committed:
Add delete service functionality to deleteJvm

In addition

- Add test cases for different delete scenarios

LinHungWu committed:
Use TreeMap and overwrite Comparator and unit test for checkState

LinHungWu committed:
Use TreeMap and overwrite Comparator and unit test for checkState

akinny committed:
added max limit to history
made max history configurable
default ui to 10,000 records
Remove History from get Group & Webserver calls

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImpl.java:
```diff
@@ -75,4 +75,16 @@ public CommandOutput call() throws Exception {
+
+    public boolean checkSameState(final JvmControlOperation jvmControlOperation, final JvmState jvmState) {
+        return (jvmControlOperation.START.toString().equals(jvmControlOperation.name()) && jvmState.isStartedState()) ||
+                (jvmControlOperation.STOP.toString().equals(jvmControlOperation.name()) && !jvmState.isStartedState()) ? true : false;
```
Remove the "? true : false" at the end of this conditional - it's already implied

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImpl.java:
```diff
@@ -75,4 +75,16 @@ public CommandOutput call() throws Exception {
+    }
+}
+
+class JvmTreeSet implements Comparator<Jvm> {
```
Rename to JvmTreeSetComparator

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupWebServerControlServiceImpl.java:
```diff
@@ -76,4 +72,8 @@ public CommandOutput call() throws Exception {
 
+    public boolean checkSameState(final WebServerControlOperation webServerControlOperation, final String webServerStateLabel) {
+        return (webServerControlOperation.START.toString().equals(webServerControlOperation.name()) && webServerStateLabel.equalsIgnoreCase(WebServerReachableState.WS_REACHABLE.toStateLabel())) ||
+                (webServerControlOperation.STOP.toString().equals(webServerControlOperation.name()) && webServerStateLabel.equalsIgnoreCase(WebServerReachableState.WS_UNREACHABLE.toStateLabel())) ? true : false;
```
Remove the "? true : false"

jmahmoodcc commented on jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImplTest.java:
```diff
@@ -48,4 +51,12 @@ public void testControlGroupWithInvalidGroup() {
     }
+
+    @Test
+    public void testcheckSameState(){
```
Add tests for the other states that return true for isStarted as well:
Initializing, initialized, start, starting, stop, stop, stopping, destroying

jmahmoodcc reviewed (changes requested):
looks good, just some minor changes

LinHungWu committed:
Jwala should only allow re-name JVM when in New State

akinny committed:
Make numOfRec configurable

jeddcuison committed:
Add delete service functionality to deleteWebServer

In addition

- Add test cases for different delete scenarios

jeddcuison committed:
Refactor

- Define variables common in several methods as class variables
- Remove unneeded setting and clearing of application properties root path

jeddcuison committed:
Add delete web server and delete JVM functionality in the group operations page

akinny committed:
Merge branch 'master' of https://github.com/cerner/jwala into resolve-large-history-issue

akinny committed:
Renamed properties

Sharvaribarve committed:
Java home set in jpajvmpersistence

jeddcuison committed:
Add $1 to query only the service being deleted

jeddcuison committed:
Change log error to debug

Sharvaribarve committed:
Addressed code review by taking care of leading/trailing white spaces and added cases for backslash

LinHungWu committed:
change pop-up message to remind the installer

rahulsayini committed:
check for existing associations before remove of media

jeddcuison committed:
Change query parameter and add test for 204

rahulsayini committed:
add a logger and edit error message

akinny committed:
Add missing default setenv.bat

rahulsayini committed:
add test case for media delete dependency

jeddcuison committed:
Log hardDelete and make it final

LinHungWu committed:
update the message

jeddcuison committed:
Change logging level to warn and error and remove duplicate comment

rahulsayini committed:
change equals to equalsIgnoreCase

jeddcuison committed:
Merge pull request #45 from cerner/JDKProperty

re-merging after fix master (previously merged as #43)

jeddcuison committed:
Merge pull request #46 from cerner/jwala-should-only-allow-rename-jvm-in-new-state

Jwala should only allow rename jvm in new state

jeddcuison committed:
Merge pull request #47 from cerner/resolve-large-history-issue

Resolve large history issue

PeterTeunissen committed:
Fixes for the hct.xml at the webapp level.

PeterTeunissen committed:
Merge pull request #48 from cerner/fix_webapp_resource_deploy

Fixes for the hct.xml at the webapp level.

Sharvaribarve committed:
Changed 12 hour to 24 hour format

LinHungWu committed:
updating buildVersion for JWALA to 0.0.185 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.186-SNAPSHOT

jmahmoodcc committed:
remove the remote.paths.httpd.conf property

jmahmoodcc committed:
Revert "remove the remote.paths.httpd.conf property"

This reverts commit 2fc7c55f3ce180c49253e5e22d28a51a0120c0f3.

jmahmoodcc committed:
Merge branch 'master' of github.com:cerner/jwala

LinHungWu committed:
Uploading non-war file in web archive of web app instead of War file------>No error thrown + file uploaded on remote server.

jmahmoodcc committed:
remove the remote.paths.httpd.conf property

jeddcuison committed:
Do not let user attempt to add a JVM if there are not groups defined

Details: Display a message that there are no groups defined and that a group needs to be added before creating JVMs

jeddcuison committed:
Fix logical error, specify the correct message and fix bug wherein the JVM table is not shown

jeddcuison committed:
Remove redundant code

jeddcuison committed:
Call doneCallback if defined

jeddcuison committed:
Do not let user attempt to add a web server if there are not groups defined

Details: Display a message that there are no groups defined and that a group needs to be added before creating web servers

jeddcuison committed:
Do not let user attempt to add a web application if there are not groups defined

Details: Display a message that there are no groups defined and that a group needs to be added before creating web applications

jeddcuison committed:
Remove "tr<space>" to make element querying faster

jeddcuison committed:
Merge pull request #42 from cerner/make-jvm-webserver-delete-consistent

Make jvm webserver delete consistent

rahulsayini committed:
change to use promisedDel and address review comments

akinny committed:
Merge pull request #51 from cerner/SortedDates

Changed 12 hour to 24 hour format

akinny committed:
Merge pull request #49 from cerner/upload-resource-war

Uploading non-war file in web archive of web app instead of War file

Sharvaribarve committed:
Changed comment format

Sharvaribarve committed:
Reformatted comments

rahulsayini committed:
revert back the changes and update MediaConfig.js confirmDeleteCallback

akinny committed:
Merge branch 'master' of https://github.com/cerner/jwala

jmahmoodcc committed:
remove references to httpdconfig in web server domain object, and replace with the deploy path from meta data only for httpd.conf resources

jmahmoodcc committed:
Merge branch 'master' of github.com:cerner/jwala

merge latest

akinny committed:
Added a label to dropdown list

akinny committed:
Merge pull request #44 from cerner/delete-media-throw-dependency-error

check for existing associations before removing media

akinny committed:
Merge branch 'master' of https://github.com/cerner/jwala

jmahmoodcc committed:
Merge pull request #39 from cerner/RESTSpecialCharacters

Incorporate backend validation of names for groups,web-servers and JVMs

jmahmoodcc committed:
remove the transactional annotation when getting jvms from a group, and use the jvm persistence layer to get jvms by group name instead

akinny committed:
Merge branch 'master' of https://github.com/cerner/jwala

akinny committed:
Fix unjar for Linux

rahulsayini committed:
Merge pull request #52 from cerner/add-label-select-jdk

Add label select jdk

akinny committed:
Review fixes

jeddcuison committed:
Only set web server state to unreachable once after a state change

In addition:

- Log the state change as a warning
- Log the state change only once
- Refactor: compare HttpStatus enum using equals

akinny committed:
Merge pull request #54 from cerner/fix-unjar-script-for-linux

Fix unjar script for linux

jmahmoodcc committed:
update deploy to use the meta data deploy path and file name

LinHungWu committed:
updating buildVersion for JWALA to 0.0.186 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.187-SNAPSHOT

jmahmoodcc committed:
remove jpa web server attributes: docRoot, errSts, httpConfigFile, svrRoot

jmahmoodcc committed:
remove the svrroot, docroot, and httpconfigfilepath from the web server table in the db

jeddcuison committed:
Make status path accept full URL or relative path

This enables users to manually specify the protocol e.g. https or http

jmahmoodcc committed:
restore previous sql commands to add web server attributes, and add versions to the version table

Sharvaribarve committed:
Added sorting for JVM list

akinny committed:
Merge pull request #55 from cerner/clean-up-webserver-ping-log-messages

Only set web server state to unreachable once after a state change

akinny committed:
Fix windows script to create jvm instance dir

akinny committed:
Make review changes

akinny committed:
Make review changes

akinny committed:
Merge pull request #59 from cerner/fix-unjar-script-for-linux

Fix cygwin script to create jvm instance dir

Sharvaribarve committed:
Sorted jvm list with Named Query

jeddcuison committed:
Merge pull request #58 from cerner/resourcesSorted

Sorted jvms by name in resources list

LinHungWu committed:
updating buildVersion for JWALA to 0.0.187 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.188-SNAPSHOT

jmahmoodcc committed:
add the web server name to the pid file name, and only remove the web server attributes from the database instead of adding them prior to removing

jmahmoodcc committed:
Merge branch 'master' of github.com:cerner/jwala

jmahmoodcc committed:
Merge branch 'master' into webserver-deploy

jmahmoodcc committed:
add username and password back to JVM table

jmahmoodcc committed:
Merge pull request #60 from cerner/webserver-deploy

The httpd.conf path should only be defined in the meta data

jmahmoodcc committed:
Merge pull request #38 from cerner/deploy-app-resources

Fix for deploy of application resources not checking for started JVMs

jmahmoodcc committed:
Merge pull request #50 from cerner/do-not-show-jvm-ws-webapp-when-there-is-no-group-defined

Do not show jvm ws webapp when there is no group defined

jmahmoodcc committed:
merge the latest from master

jmahmoodcc committed:
Merge pull request #57 from cerner/webserver-status-uri-rework

Make status path accept full URL or relative path

LinHungWu committed:
updating buildVersion for JWALA to 0.0.188 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.189-SNAPSHOT

LinHungWu committed:
Found the root cause (duplicated jvms from group.getjvms) and fix Media object for equals and hashcode. Add more unit test for all jvm state and webserver state.

LinHungWu committed:
Found the root cause (duplicated jvms from group.getjvms) and fix Media object for equals and hashcode. Add more unit test for all jvm state and webserver state.

LinHungWu committed:
Merge branch 'fix-jvm-webserver-group-start-stop' of github.com:cerner/jwala into fix-jvm-webserver-group-start-stop

Conflicts:
	jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImpl.java
	jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImplTest.java

LinHungWu committed:
Remove parentGroup from Jvm and WebServer object since we use group set to link Jvm and WebServer.

LinHungWu committed:
Remove parentGroup from Jvm and WebServer object since we use group set to link Jvm and WebServer.

LinHungWu committed:
pull from master and resolved the conflict.

LinHungWu committed:
Merge branch 'fix-jvm-webserver-group-start-stop' of github.com:cerner/jwala into fix-jvm-webserver-group-start-stop

Conflicts:
	jwala-common/src/main/java/com/cerner/jwala/common/domain/model/webserver/WebServer.java
	jwala-common/src/test/groovy/com/cerner/jwala/template/TestResourceFileGenerator.groovy
	jwala-services/src/test/java/com/cerner/jwala/service/balancermanager/MockGroup.java
	jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupJvmControlServiceImplTest.java
